tfe/resource_tfe_workspace.go

```diff
@@ -671,15 +676,54 @@ func resourceTFEWorkspaceDelete(d *schema.ResourceData, meta interface{}) error
 	id := d.Id()
 
 	log.Printf("[DEBUG] Delete workspace %s", id)
-	err := tfeClient.Workspaces.DeleteByID(ctx, id)
+
+	ws, err := tfeClient.Workspaces.ReadByID(ctx, id)
 	if err != nil {
 		if err == tfe.ErrResourceNotFound {
 			return nil
 		}
 		return fmt.Errorf(
-			"Error deleting workspace %s: %w", id, err)
+			"Error reading workspace %s: %w", id, err)
+	}
+
+	forceDelete := d.Get("force_delete").(bool)
+
+	// presence of Permissions.CanForceDelete will determine if current version of TFE supports safe deletes
+	if ws.Permissions.CanForceDelete == nil {
+		if forceDelete {
+			err = tfeClient.Workspaces.DeleteByID(ctx, id)
+		} else {
+			return fmt.Errorf(
+				"Error deleting workspace %s: This workspace must be force deleted by setting force_delete=true", id)
+		}
+	} else if *ws.Permissions.CanForceDelete {
+		if forceDelete {
+			err = tfeClient.Workspaces.DeleteByID(ctx, id)
+		} else {
+			err = tfeClient.Workspaces.SafeDeleteByID(ctx, id)
+			if err != nil {
+				if strings.Contains(err.Error(), "conflict") {
```

Comment:
If possible, I might go with a `strings.HasPrefix(err.Error(), "conflict")` here, just to be more selective in the string matching.... just so we dont end up matching against something like "The workspace i-put-conflict-in-the-name-of-my-workspace doesnt exist"

Reply:
that makes total sense! this felt wrong, thanks for pointing me to `hasPrefix` !